frontend/src/pages/AllCategories.tsx

```diff
@@ -0,0 +1,97 @@
+import { useQuery } from '@apollo/client';
+import { Stack, Typography, Button } from '@mui/material';
+import { DataContext } from 'contexts';
+import { useContext } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+import { StringBank } from 'strings';
+import { IAgreement } from 'types';
+import { agreementsQuery } from 'utils/queries';
+import AddIcon from '@mui/icons-material/Add';
+import img from '../assets/Group_120.png';
+import { AgreementCarousel } from 'components';
+import { useTranslation } from 'react-i18next';
+import EmptyCategoryPage from './EmptyCategoryPage';
+import { GroupContext } from 'contexts/group';
+
+const AllCategories = () => {
+  const navigate = useNavigate();
+  const { t } = useTranslation();
+  let { categoryId } = useParams();
+  const { user } = useContext(DataContext);
+  const { slug } = useContext(GroupContext);
+  if (categoryId) {
+    categoryId = categoryId.substring(0, categoryId.length - 1);
+  }
+  const currentGroup = user?.groups?.find((group) => group.slug === slug);
+  const vars = {
+    groupId: currentGroup?.id || -1,
+    isArchived: false,
+    categoryId,
+  };
+  const { data: categoryAgreements } = useQuery<{
+    core_agreements: IAgreement[];
+  }>(agreementsQuery(categoryId), {
+    variables: vars,
+  });
+  const handleMenuItemClick = (e: React.MouseEvent<HTMLElement>, slug = '') => {
+    navigate(`/${slug}/new-agreement`);
+  };
+
+  const agreements = categoryAgreements;
+  const categoryName = agreements?.core_agreements[0]?.category?.name || 'category';
+  if (agreements && agreements?.core_agreements.length > 0) {
```

Comment:
Let's make this less nested (see comment on archive page)

Reply:
done